tests/Hedgehog.CSharp.Tests/NameTests.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Xunit;
+
+namespace Hedgehog.CSharp.Tests
+{
+    public class NameTests
+    {
+        private static Type[] _publicApiTypes =
+            { typeof(Gen)
+            , typeof(Range)
+            , typeof(Property)
+            };
+
+        public static IEnumerable<MemberInfo[]> AllPublicMembers()
+        {
+            var bindingFlags =
+                BindingFlags.Public |
+                BindingFlags.Instance |
+                BindingFlags.Static |
+                BindingFlags.DeclaredOnly;
+
+            foreach (var type in _publicApiTypes)
+            {
+                var members = type.GetMembers(bindingFlags);
+
+                foreach (var member in members)
+                {
+                    // Ignore special members like indexers etc.
```

Comment:
The body the `foreach` loop requires features not found in the C# compiler that ships with Visual Studio 2015. We could either refactor the body of this loop to plain-old C# or do something like [this](https://github.com/hedgehogqa/fsharp-hedgehog/pull/113#issuecomment-341953738), so that the project compiles.

Reply:
Changed this to C#6-compatible code and forced the project to only use C#6.

Reply:
@Porges, @jystic runs into this on his local:

```
LinqTests.cs(4,23): error CS0305: Using the generic type `Hedgehog.Property<a>' requires `1' type argument(s)
```

It's basically that `using static Hedgehog.Property;` one we were talking about yesterday. You were right that we have to get rid of it (it *does* compile in VS2015 but has issue with Mono)...

Reply:
I guess, we should probably leave this as-is, and have @jystic install C# 6 on his machine. I think it's fair to have C# 6 as base...

Reply:
C#6 is 2015 so yes 😁